src/main/java/com/pickyfy/pickyfy/service/UserServiceImpl.java

```diff
@@ -42,16 +43,16 @@ public UserInfoResponse getUser(String accessToken){
     }
 
     @Transactional
-    public UserUpdateResponse updateUser(String accessToken, UserUpdateRequest request){
+    public Long updateUser(String accessToken, UserUpdateRequest request, MultipartFile image){
         User user = getAuthenticatedUser(accessToken);
 
         User updatedUser = user.toBuilder()
                 .nickname(request.nickname() != null ? request.nickname() : user.getNickname())
-                .profileImage(request.profileImage() != null ? request.profileImage() : user.getProfileImage())
+                .profileImage(s3Service.upload(image) != null ? s3Service.upload(image) : user.getProfileImage())
```

Comment:
이 부분에서 s3Service.upload(image)를 불필요하게 두 번 호출하고있는 것 같습니다! 아래와 같이 optional을 사용하거나 변수선언을 해서 중복 업로드 과정을 없애면 좋을 것 같습니다
```suggestion
.profileImage(Optional.ofNullable(s3Service.upload(image)).orElse(user.getProfileImage()))
```

Reply:
지금 보니까 s3Service.upload(image)에서 예외가 발생할 가능성은 있지만 null이 반환될 가능성은 없어보이네요 따라서 Optional을 사용하지 않으면서 중복호출을 막는게 좋을 것 같아요! 이러면 아래 리뷰도 해결할 수 있을 것 같습니다
```suggestion
                .profileImage(image != null ? s3Service.upload(image) : user.getProfileImage())
```

Reply:
> 이 부분에서 s3Service.upload(image)를 불필요하게 두 번 호출하고있는 것 같습니다! 아래와 같이 optional을 사용하거나 변수선언을 해서 중복 업로드 과정을 없애면 좋을 것 같습니다

좋은 방법인것 같습니다.

Reply:
> 지금 보니까 s3Service.upload(image)에서 예외가 발생할 가능성은 있지만 null이 반환될 가능성은 없어보이네요 따라서 Optional을 사용하지 않으면서 중복호출을 막는게 좋을 것 같아요! 이러면 아래 리뷰도 해결할 수 있을 것 같습니다

넹 요렇게 하겠습니다